### DISCUSSION TITLE
Switch back `_joinedOnce` to `false` after leaving the channel

### DISCUSSION THREAD

**Discussion by TesteurManiak** [braverhealth/phoenix-socket-dart#109] [Q&A]
The `PhoenixChannel._joinedOnce` property is only set to `false` when the class is instanciated, I think it would be expected for it to be also set after calling `PhoenixChannel.leave()`.

**Comment by jaybe78** [comment]
Any progress on this ?

**Comment by jaybe78** [comment]
Hi @matehat,

As @TesteurManiak pointed out the `_joinedOnce` is set to true in the `join` function but never set back to false in `leave` and `close`.

it does not seem correct...

Do you have any clue why ?

Cheers

**Comment by matehat** [comment]
When you `.leave()` a channel, the expectation is that you would remove it from the socket as well. So something like:

```dart
final push = channel.leave();
socket.removeChannel(channel);
await push.future;
```

After that, if you need to re-open the same channel, you'd do something like:

```dart
final channel = socket.addChannel("some-topic");
final push = channel.join();
await push.future;
```

If you've removed the channel beforehand, the new channel is brand new so you don't have to worry about the `_joinedOnce` again.

**Reply by jaybe78** [reply]
Thanks @matehat.
